modules/auxiliary/scanner/ssl/ssl_version.rb

```diff
@@ -0,0 +1,463 @@
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed certificate expired",
+        refs: references,
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # not yet valid
+    if cert.not_before > DateTime.now
+      print_good("Certificate not yet valid: #{cert.not_after}")
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed certificate not yet valid",
+        refs: references,
+        exploited_at: Time.now.utc
+      )
+    end
+  end
+
+  # Fingerprint a single host
+  def run_host(ip)
+    versions = get_ssl_versions
+
+    certs_found = {}
+    skip_ssl_version = false
+    vprint_status("Scanning #{ip} for: #{versions.map(&:to_s).join(', ')}")
+
+    versions.each do |version|
+      skip_ssl_version = false
+      ciphers = get_ssl_cipher_suites(version)
+      vprint_status("Scanning #{ip} #{version} with ciphers: #{ciphers.map(&:to_s).join(', ')}")
+      ciphers.each do |cipher|
+        break if skip_ssl_version
+
+        vprint_status("Attempting connection with SSL Version: #{version}, Cipher: #{cipher}")
+        begin
+          # setting the connect global to false means we can't see the socket, therefore the cert
+          connect(true, { 'SSL' => true, 'SSLVersion' => version.sub('.', '_').to_sym, 'SSLCipher' => cipher }) # Force SSL
+          print_good("Connected with SSL Version: #{version}, Cipher: #{cipher}")
+
+          if sock.respond_to? :peer_cert
+            cert = OpenSSL::X509::Certificate.new(sock.peer_cert)
+            # https://stackoverflow.com/questions/16516555/ruby-code-for-openssl-to-generate-fingerprint
+            cert_fingerprint = OpenSSL::Digest::SHA1.new(cert.to_der).to_s
+            if certs_found.key? cert_fingerprint
+              # dont check the cert more than once if its the same cert
+              check_vulnerabilities(ip, version, cipher, nil)
+            else
+              loot_cert = store_loot('ssl.certificate', 'text/plain', ip, cert.to_text)
+              print_good("Certificate saved to loot: #{loot_cert}")
+              print_cert(cert, ip)
+              check_vulnerabilities(ip, version, cipher, cert)
+            end
+            certs_found[cert_fingerprint] = cert
+          end
+        rescue ::OpenSSL::SSL::SSLError => e
+          error_message = e.message.match(/ state=(.+)$/)
+
+          if error_message.nil?
+            vprint_error("\tSSL Connection Error: #{e}")
+            next
+          end
+
+          # catch if the ssl_version/protocol isn't allowed and then we can skip out of it.
+          if error_message[1].include? 'no protocols available'
+            skip_ssl_version = true
+            vprint_error("\tDoesn't accept #{version} connections, Skipping")
+            break
+          end
+          vprint_error("\tDoes not accept #{version}, error message: #{error_message[1]}")
```

Comment:
I thought this was meant to be printing out the cipher given that by this point I thought we determined that the SSL protocol itself is accepted but maybe the cipher isn't supported? This is also supported by your comment at the top of this code block so I'm a bit confused here.